src/generate_conf.rs

```diff
@@ -76,19 +77,39 @@ fn generate_config(
 ) -> Result<(Vec<Interface>, NetworkConfig), anyhow::Error> {
     let network_state = NetworkState::new_from_yaml(&data)?;
 
-    let interfaces = extract_interfaces(&network_state);
-    validate_interfaces(&interfaces, require_mac_addresses)?;
-
     let config = network_state
         .gen_conf()?
         .get("NetworkManager")
         .ok_or_else(|| anyhow!("Invalid NM configuration"))?
         .to_owned();
 
+    let interfaces = extract_interfaces(&network_state, &config);
+    validate_interfaces(&interfaces, require_mac_addresses)?;
+
     Ok((interfaces, config))
 }
 
-fn extract_interfaces(network_state: &NetworkState) -> Vec<Interface> {
+fn extract_interfaces(
+    network_state: &NetworkState,
+    config_files: &NetworkConfig,
+) -> Vec<Interface> {
+    let mut config_files_map: HashMap<String, Vec<String>> = HashMap::new();
+
+    for (_filename, content) in config_files {
+        let mut config = Ini::new();
+        config
+            .read(content.to_string())
+            .expect("Unable to read nmconnection file");
```

Comment:
We cannot afford to essentially "panic" here. If we agree on the suggestion above, we can extract this parsing in a separate function which can then propagate the error up the chain.

Reply:
I've extracted this to a separate function. How to properly propagate I need some guidance.